airflow-core/src/airflow/dag_processing/processor.py

```diff
@@ -94,6 +96,21 @@ def _parse_file_entrypoint():
 
 def _parse_file(msg: DagFileParseRequest, log: FilteringBoundLogger) -> DagFileParsingResult | None:
     # TODO: Set known_pool names on DagBag!
+
+    # Pre-import modules
+    # Read the file to pre-import airflow modules used.
+    # This prevents them from being re-imported from zero in each "processing" process
+    # and saves CPU time and memory.
+
+    if conf.getboolean("scheduler", "parsing_pre_import_modules", fallback=True):
+        for module in iter_airflow_imports(msg.file):
+            try:
+                importlib.import_module(module)
+            except Exception as e:
```

Comment:
```suggestion
            except Exception as e:
```

We probably should use concrete exception here instead